src/test/java/uk/gov/justice/digital/hmpps/keyworker/services/ComplexityOfNeedServiceTest.kt

```diff
@@ -16,43 +19,86 @@ import uk.gov.justice.digital.hmpps.keyworker.events.ComplexityOfNeedListenerTes
 @ExtendWith(MockitoExtension::class)
 class ComplexityOfNeedServiceTest {
 
+  companion object {
+    const val OFFENDER_NO = "A12345"
+    const val OFFENDER_NO_1 = "A12345"
+    const val OFFENDER_NO_2 = "A12346"
+    const val OFFENDER_NO_3 = "A12347"
+
+    val ENABLED_PRISONS = setOf("MDI")
+  }
+
   @Mock
   lateinit var keyworkerService: KeyworkerService
 
   @Mock
   lateinit var telemetryClient: TelemetryClient
 
-  lateinit var complexityOfNeedService: ComplexityOfNeedService
+  @Mock
+  lateinit var complexityOfNeedAPI: ComplexityOfNeedAPI
 
-  companion object {
-    const val OFFENDER_NO = "A12345"
-  }
+  lateinit var complexityOfNeedService: ComplexityOfNeedService
 
   @BeforeEach
   fun setUp() {
-    complexityOfNeedService = ComplexityOfNeedService(keyworkerService, telemetryClient)
+    complexityOfNeedService =
+      ComplexityOfNeedService(keyworkerService, complexityOfNeedAPI, ENABLED_PRISONS, telemetryClient)
   }
 
   @Test
   fun `should not deallocate offenders that do not have high complexity of needs`() {
-    complexityOfNeedService.onComplexityChange(OFFENDER_NO, ComplexityOfNeedLevel.LOW)
+    complexityOfNeedService.onComplexityChange(OFFENDER_NO_1, ComplexityOfNeedLevel.LOW)
 
-    verify(keyworkerService, never()).deallocate(OFFENDER_NO)
+    verify(keyworkerService, never()).deallocate(OFFENDER_NO_1)
   }
 
   @Test
   fun `should deallocate offenders that have high complexity of needs`() {
-    complexityOfNeedService.onComplexityChange(OFFENDER_NO, ComplexityOfNeedLevel.HIGH)
+    complexityOfNeedService.onComplexityChange(OFFENDER_NO_1, ComplexityOfNeedLevel.HIGH)
+
+    verify(keyworkerService, times(1)).deallocate(OFFENDER_NO_1)
+  }
+
+  @Test
+  fun `should not filter out complex offenders for none enabled prisons`() {
+    val noneComplexOffenders = complexityOfNeedService.getComplexOffenders("LEI", setOf(OFFENDER_NO_1))
```

Comment:
This is confusing... maybe better to name the variable "noneEnabledPrisonOffenders" or something.

Actually this indicates that there won't be any auto-allocation of offenders in prisons that aren't enabled - is that right?